docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,164 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
+
+**s2n_offered_psk_get_identity** gets the PSK identity and PSK identity length for a given offered PSK object.
+Callers can use this API to obtain the identity value set on the **s2n_offered_psk** object.
+
+**s2n_offered_psk_list_has_next** checks whether the offered PSK list has an offered psk object next in line in the list.
+An offered PSK list contains all the pre-shared keys offered by the client for the server to select.
+If an offered PSK exists next in line, this API returns True, otherwise False.
+Use this API prior to calling **s2n_offered_psk_list_next** to obtain the next offered PSK object from the offered PSK list.
+
+**s2n_offered_psk_list_next** obtains the next offered PSK object from the offered PSK list.
+Use **s2n_offered_psk_list_has_next** prior to this API call to determine if an offered PSK exists next in the list.
+
+**s2n_offered_psk_list_reread** returns the offered PSK list to its original read state.
+Calls to **s2n_offered_psk_list_next** will start over from the beginning of the list.
+
+**s2n_offered_psk_list_choose_psk** selects a PSK from the offered PSK list to be used for the connection.
+This API matches the PSK identity received from the client against the server's known PSK identities list,
+in order to choose the PSK to be used for the connection. If the PSK identity sent from the client is NULL,
+no PSK is chosen for the connection. If the client offered PSK identity has no matching PSK identity with the server,
+an error will be returned. Use this API along with the **s2n_psk_selection_callback** callback to select a PSK for the connection.
+
+**s2n_psk_selection_callback** is a callback function to select a PSK from a list of offered pre-shared keys.
+Use this callback to implement custom PSK selection logic. By default s2n-tls server chooses the first identity in its PSK
+list that also appears in the client's PSK list.
+
+**s2n_config_set_psk_selection_callback** sets the callback to select the matching PSK . If this callback is not set,
+the server chooses the first identity in its PSK list that also appears in the client's PSK list.
+
+For Example:
+
+```c
+int s2n_psk_selection_callback(struct s2n_connection *conn, void *context,
+        struct s2n_offered_psk_list *psk_identity_list)
+{
+    uint16_t *wire_index_choice = (uint16_t*) context;
+
+    struct s2n_offered_psk offered_psk = { 0 };
+    uint16_t idx = 0;
+    while(s2n_offered_psk_list_has_next(psk_identity_list)) {
+        POSIX_GUARD(s2n_offered_psk_list_next(psk_identity_list, &offered_psk));
+        if (idx == *wire_index_choice) {
+            POSIX_GUARD(s2n_offered_psk_list_choose_psk(psk_identity_list, &offered_psk));
+            break;
+        }
+        idx++;
+    };
+    return S2N_SUCCESS;
+}
+
+struct s2n_config *config = s2n_config_new();
+uint16_t expected_wire_choice = 0;
+s2n_config_set_psk_selection_callback(config, s2n_psk_selection_callback, &expected_wire_choice));
```

Comment:
This part of the example is probably unnecessary.

Reply:
It shows the use of `s2n_config_set_psk_selection_callback` and ties it together IMO. We aren't loosing anything by adding it. :)